File: testcases/paper/Methods/Methods.go
package methods

import "fmt"

func outsideMethod() {
    fmt.Println("Outside method called.")
}

type Methods struct {}

func (m Methods) methodOne() {
    fmt.Println("Method one called.")
}

func (m Methods) methodTwo() {
    fmt.Println("Method two called.")
    m.methodOne() // Calls methodOne
}

func (m Methods) methodThree() {
    fmt.Println("Method three called.")
    outsideMethod() // Calls outside method
}

File: testcases/paper/Methods/Methods.java
public class Methods {
    public void methodOne() {
        System.out.println("Method one called.");
    }

    public void methodTwo() {
        System.out.println("Method two called.");
        methodOne(); // Calls methodOne
    }

    public void methodThree() {
        System.out.println("Method three called.");
        OutsideClass.outsideMethod(); // Calls outside method
    }

    public static void main(String[] args) {
        return;
    }
}
